src/react/components/NativeBlockFactory.tsx

```diff
@@ -18,35 +18,49 @@
 
 import React from 'react';
 import _ from 'lodash';
-import { AbstractReactFactory } from '@projectstorm/react-canvas-core';
+import { AbstractReactFactory, BaseModel } from '@projectstorm/react-canvas-core';
+import { DiagramEngine } from '@projectstorm/react-diagrams';
+import type { AstarteBlock } from 'astarte-client';
+
 import NativeBlockModel from '../models/NativeBlockModel';
 import NativeBlockWidget from './NativeBlockWidget';
 
-class NativeBlockFactory extends AbstractReactFactory {
-  constructor(blockDefinitions) {
-    super('astarte-native');
+type GenerateReactWidgetEvent = Parameters<AbstractReactFactory['generateReactWidget']>['0'];
```

Comment:
This can also be done using `GenerateWidgetEvent` from the '@projectstorm/react-canvas-core' library
see https://github.com/projectstorm/react-diagrams/blob/master/packages/diagrams-demo-gallery/demos/demo-custom-link-label/EditableLabelFactory.tsx

Reply:
Nice catch, thanks